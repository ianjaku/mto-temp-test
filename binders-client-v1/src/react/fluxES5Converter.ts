/* eslint-disable @typescript-eslint/no-explicit-any */
// eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types
export function fixES5FluxContainer<T extends new (...args: any[]) => any>(
    containerClass: T
): T {
    const tmp = containerClass;
    const newClass = function(...args: any[]) {
        return new tmp(...args);
    } as unknown as T;
    newClass.prototype = tmp.prototype;
    (newClass as any).getStores = (tmp as any).getStores;
    (newClass as any).calculateState = (tmp as any).calculateState;
    return newClass;
}
