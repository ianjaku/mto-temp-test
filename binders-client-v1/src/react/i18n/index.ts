import i18n from "../../i18n";
export { useTranslation, withTranslation } from "react-i18next";
export { type TFunction } from "../../i18n";
export default i18n;
