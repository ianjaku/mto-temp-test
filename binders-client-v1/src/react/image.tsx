/* eslint-disable @typescript-eslint/explicit-module-boundary-types */
import * as React from "react";
import { pick, replace } from "ramda";
import { isVideoURL } from "../clients/imageservice/v1/visuals";

export type Unit = "%" | "vw";

// eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types
export function getFitCropStyleObject(fitBehaviour: string, aspectRatio: number, unit: Unit = "%") {
    const isLandscape = aspectRatio > 1;
    const addUnit = (amount: number) => `${amount}${unit}`;
    if (fitBehaviour === "fit") {
        if (isLandscape) {
            return {
                width: addUnit(100)
            };
        } else {
            const style = {
                height: addUnit(100),
                width: addUnit(Math.round(100 * aspectRatio))
            };
            return style;
        }
    } else {
        if (isLandscape) {
            return {
                left: addUnit(50),
                top: addUnit(50),
                height: addUnit(100),
                width: "auto",
                transform: `translate(-${addUnit(50)},-${addUnit(50)})`,
                position: "absolute"
            };
        } else {
            return {
                width: addUnit(100)
            };
        }
    }
}


export interface ImageProps {
    // eslint-disable-next-line @typescript-eslint/no-explicit-any
    image?: any;
    isOver?: boolean;
    dirtyDownSize?: boolean;
    className: string;
}

export interface ImageState {
    bgColor: string;
    width?: number;
    height?: number;
}

export default class Image extends React.Component<ImageProps, ImageState> {

    constructor(props) {
        super(props);
        this.state = {
            bgColor: props.image ? props.image.bgColor : "",
        };
    }

    handleImageLoaded(e) {
        this.setState({
            bgColor: this.props.image.bgColor,
            width: e.target.naturalWidth,
            height: e.target.naturalHeight
        });
    }

    dirtyDownsizeImageImg(img) {
        let url;
        if (img.url) {
            url = img.url;
        } else {
            url = img;
        }
        let replaceTarget = "/medium";
        if (isVideoURL(url)) {
            replaceTarget = "/video_screenshot";
        }
        return replace(/\/original/, replaceTarget, url);
    }

    getImageSrc() {
        if (this.props.dirtyDownSize) {
            return this.dirtyDownsizeImageImg(this.props.image);
        }
        if (this.props.image.buildRenderUrl) {
            return this.props.image.buildRenderUrl();
        }
        if (this.props.image.medium) {
            return this.props.image.medium;
        }
        return this.props.image;
    }

    getImageProps() {
        const { rotation }  = this.props.image;
        // eslint-disable-next-line @typescript-eslint/no-explicit-any
        const imageProps: any = {
            ...pick(["style", "alt"], this.props),
            className: this.props.isOver ? "hovering" : "",
            src: this.getImageSrc(),
            onLoad: this.handleImageLoaded.bind(this)
        };
        let rotationStyle = {};
        imageProps.style = Object.assign(imageProps.style ? imageProps.style : {}, this.getFitCropStyling());
        if(rotation) {
            const rotate = `rotate(${rotation}deg)`;
            rotationStyle = {transform:  (imageProps.style.transform) ? `${imageProps.style.transform} ${rotate}` : rotate};
            imageProps.style = {...imageProps.style, ...rotationStyle};
        }
        return imageProps;
    }

    getFitCropStyling() {
        return getFitCropStyleObject(this.props.image.fitBehaviour, this.getAspectRatio());
    }

    getAspectRatio() {
        return this.state.height ?
            this.state.width / this.state.height :
            1;
    }

    render() {
        const isTransparent = this.props.image.bgColor === "transparent";
        const wrapperStyle = { backgroundColor: isTransparent ? "inherit" : `#${this.props.image.bgColor}` } as React.CSSProperties;

        return (
            <div className={`${this.props.className ? this.props.className : ""} image-wrapper ${(isTransparent ? " transparent" : "")}`} style={wrapperStyle}>
                <div className="image-container">
                    <img {...this.getImageProps() } />
                </div>
            </div>
        );
    }

}
