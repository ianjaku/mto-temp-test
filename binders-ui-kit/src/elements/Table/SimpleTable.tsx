import * as React from "react";
import * as ReactDomServer from "react-dom/server";
import Table, { ITableProps } from "./Table";
import { filterExportOnlyColumns, prepareDatesFormats, prepareHeaders, sortFunctions } from "./utils";
import { fmtDateIso8601TZ, fmtDateIso8601TimeLocalizedTZ } from "@binders/client/lib/util/date";
import { SORT } from "./types";
import { equals } from "ramda";
import { exportRowsToSheetsFiles } from "@binders/client/lib/util/xlsx";
import { isDate } from "date-fns";

export interface IDataTableState {
    sort: number[];
    page: number;
    // eslint-disable-next-line @typescript-eslint/no-explicit-any
    data: any;
    recordsPerPage: number;
    filterEnabled: boolean;
    filtered?: boolean;
}
// to override type in interface
interface ISimpleTableProps extends ITableProps {
    customHeaders?: Array<{ label, type, format?, defaultSort?, exportOnly?} | string>;
    filtered?: boolean;
    searchable?: boolean;
    exportable?: boolean;
    defaultSort?: { columnIndex: number, order: SORT };
    onExportData?: (type: string) => void;
    // eslint-disable-next-line @typescript-eslint/no-explicit-any
    exportDataTransformFn?: (data: any) => any;
}

function SimpleTable(TableComponent) {
    return class extends React.Component<ISimpleTableProps, IDataTableState> {
        public static defaultProps: Partial<ISimpleTableProps> = {
            data: [],
            filtered: false,
            onChangeRecordsPerPage: () => {
                return;
            },
            onPageChange: () => {
                return;
            },
            onSort: () => {
                return;
            },
            searchable: false,
        };

        private normalizedHeaders: Array<{ label, type, format?}>;
        private exportHeaders: Array<{ label, type, format?}>;

        constructor(props) {
            super(props);
            const { customHeaders, defaultSort, data, recordsPerPage } = this.props;
            const renderableHeaders = customHeaders.filter(header => typeof header === "string" || !header.exportOnly);
            this.normalizedHeaders = customHeaders ? prepareHeaders(renderableHeaders, data[0]) : [];
            this.exportHeaders = customHeaders ? prepareHeaders(customHeaders, data[0]) : [];
            this.onSearch = this.onSearch.bind(this);
            this.onExportData = this.onExportData.bind(this);
            const sort = Array(this.normalizedHeaders.length).fill(SORT.NOSORT);
            if (defaultSort) {
                const { columnIndex, order } = defaultSort;
                sort[columnIndex] = order;
            }
            this.state = {
                data,
                filterEnabled: false,
                filtered: props.filtered,
                page: 1,
                recordsPerPage: recordsPerPage || 10,
                sort,
            };
        }

        componentDidMount() {
            const { data, filtered } = this.props;
            this.setState({
                data,
                filtered,
                page: 1,
            });
        }

        componentDidUpdate(prevProps, prevState) {
            const { data: prevData } = prevProps;
            const { data, filtered } = this.props;
            const { page } = this.state;
            const { filterEnabled: prevFilterEnabled } = prevState;
            if (!prevFilterEnabled && !equals(prevData, data)) {
                this.setState({
                    data,
                    filtered,
                    page: (filtered || data.length <= this.state.recordsPerPage) ? 1 : page,
                });
            }
        }

        public onPageChange = page => {
            this.props.onPageChange(page);
            this.setState({ page });
        }

        public onSort = sort => {
            this.props.onSort(sort);
            this.setState({ sort });
        }


        public getSortedData(filterExportOnly = false) {
            let data = [...this.state.data];
            const { customHeaders } = this.props;
            const sort = [...this.state.sort];
            for (let headKey = 0; headKey < customHeaders.length; headKey++) {
                // eslint-disable-next-line no-prototype-builtins
                if (customHeaders[headKey] && customHeaders[headKey].hasOwnProperty("exportOnly")) {
                    // you cant sort by exportOnly column ( not visible on UI )
                    // fix sort map on the fly then
                    sort.splice(headKey, 0, 0);
                    continue;
                }
                if (sort[headKey] !== SORT.NOSORT) {
                    const header = this.exportHeaders[headKey];
                    const sortFn = header && sortFunctions[header.type] || (() => -1);
                    data = data.sort((c, d) => {
                        // eslint-disable-next-line no-prototype-builtins
                        const item1 = c[headKey] && c[headKey].hasOwnProperty("value") ? c[headKey].value : c[headKey];
                        // eslint-disable-next-line no-prototype-builtins
                        const item2 = d[headKey] && d[headKey].hasOwnProperty("value") ? d[headKey].value : d[headKey];
                        return sortFn(sort[headKey], item1, item2);
                    });

                }
            }
            const result = prepareDatesFormats(this.normalizedHeaders, data);
            return filterExportOnly ? filterExportOnlyColumns(result) : result;
        }

        public onChangeRecordsPerPage = amount => {
            this.props.onChangeRecordsPerPage(amount);
            this.setState({ recordsPerPage: amount, page: 1 });
        }

        public render() {
            const { exportable, searchable, ...tableProps } = this.props;
            const { page, recordsPerPage, sort } = this.state;

            const start = recordsPerPage * (page - 1);
            const end = recordsPerPage * (page - 1) + recordsPerPage;
            return (
                <TableComponent
                    {...tableProps}
                    normalizedHeaders={this.normalizedHeaders}
                    max={this.state.data.length}
                    recordsPerPage={recordsPerPage}
                    onSort={this.onSort}
                    onPageChange={this.onPageChange}
                    onChangeRecordsPerPage={this.onChangeRecordsPerPage}
                    data={this.getSortedData(true).slice(start, end)}
                    onSearch={searchable ? this.onSearch : undefined}
                    onExportData={exportable ? this.onExportData : undefined}
                    sort={sort}
                />
            );
        }

        private onSearch(query) {
            const { data, onSearch } = this.props;
            if (typeof onSearch === "function") {
                onSearch(query);
                this.setState({ page: 1, sort: [] });
                return undefined;
            }
            const q = query.length > 0 ? query.toLowerCase() : false;
            const filteredData = data.reduce((out, column) => {
                const hasMatch = column.findIndex((field) => {
                    let fieldString = field && field.value ?
                        field.value :
                        field;
                    if (React.isValidElement(fieldString)) {
                        // you can pass react element to the table
                        // so if we detect one
                        // try to get html from it and strip html tags to have only the text
                        fieldString = ReactDomServer.renderToStaticMarkup(fieldString).replace(/(<([^>]+)>)/ig, "")
                    }
                    return fieldString && fieldString.toString().toLowerCase().indexOf(q) >= 0;
                }) >= 0;
                if (hasMatch || !q) {
                    out.push(column);
                }
                return out;
            }, []);
            this.setState({
                data: filteredData,
                filterEnabled: !!q,
                page: 1,
                sort: [],
            });
        }

        private onExportData(type: string): void {
            const { onExportData } = this.props;
            if (typeof onExportData === "function") {
                return onExportData(type);
            }
            const data = this.buildExportData();
            const isCsv = (type !== "excel");
            const ext = isCsv ? "csv" : "xlsx";
            const name = `exported_table_${fmtDateIso8601TZ(new Date())}.${ext}`;
            exportRowsToSheetsFiles(
                data,
                "SheetJS",
                name,
                isCsv,
            );
        }

        // eslint-disable-next-line @typescript-eslint/no-explicit-any
        private buildExportData(): Array<Array<any>> {
            const { exportDataTransformFn } = this.props;
            const data = this.getSortedData();
            const exportData = exportDataTransformFn ? exportDataTransformFn(data) : data;
            return [
                this.exportHeaders.map(h => h.label),
                ...exportData.map(row => row.map(field => {
                    const properField = field?.exportValue ?? field;
                    return isDate(properField) ? fmtDateIso8601TimeLocalizedTZ(properField) : properField;
                }),
                ),
            ];
        }
    };
}

// eslint-disable-next-line @typescript-eslint/no-explicit-any
export default SimpleTable(Table) as any;
