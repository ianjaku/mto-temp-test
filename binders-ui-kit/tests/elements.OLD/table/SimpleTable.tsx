/* eslint-disable @typescript-eslint/no-unused-vars */
import * as React from "react";
import SimpleTable from "../../../src/elements/Table/SimpleTable";
import { create } from "react-test-renderer";
import { mount } from "enzyme";
import wrapWithTheme from "../../themeHelper";

const data = [];

for (let i = 1; i < 1001; i++) {
    data.push([`John Doe${i}`, `johndoe${i}@manual.to`, i, <button>Test</button>]);
}

const headers = ["Name", "E-mail", "ID", "actions"];

const table = wrapWithTheme(<SimpleTable data={data} customHeaders={headers} recordsPerPage={5} />);
const tableWithCallbacks = (sortCb, recordsCb, pageChangeCb) => (
    wrapWithTheme(<SimpleTable
        data={data}
        customHeaders={headers}
        onChangeRecordsPerPage={recordsCb}
        onSort={sortCb}
        onPageChange={pageChangeCb}
    />)
);
// most preferred
describe("Table", () => {
    test("Table (snapshot)", () => {
        const modal = create(table);
        const serialized = modal.toJSON();
        expect(serialized).toMatchSnapshot();
    });

    // test last
    test("Should have 1000 elements and 5 per row", () => {
        const element = mount(table);
        let tableElement = element.find("table > tbody > tr");
        expect(tableElement.length).toEqual(5);

        element
            .find(".pagination > li")
            .at(8)
            .simulate("click");
        expect(
            element
                .find(".pagination > li")
                .at(6)
                .text(),
        ).toEqual("200");
        tableElement = element.find("table > tbody > tr");
        expect(tableElement.length).toEqual(5);
    });

    test("Sorting (ASC)", () => {
        const element = mount(table);
        const tableElement = element.find("table > tbody > tr");
        let intField = element
            .find("table > tbody > tr")
            .at(0)
            .find("td")
            .at(2)
            .text();
        let sortIntFieldElement = element.find("table > thead > tr th").at(2);

        // not sorted
        expect(intField).toEqual("1");
        expect(sortIntFieldElement.text().trim()).toEqual("ID");

        // sort ASC
        sortIntFieldElement.simulate("click");
        intField = element
            .find("table > tbody > tr")
            .at(0)
            .find("td")
            .at(2)
            .text();
        sortIntFieldElement = element.find("table > thead > tr th").at(2);
        expect(intField).toEqual("1");
        expect(sortIntFieldElement.text()).toEqual("IDarrow_upward");

        // next 3 pages should also be in downward trend
        const nextButton = element.find(".pagination > li").at(7);
        nextButton.simulate("click");
        nextButton.simulate("click");
        nextButton.simulate("click");
        intField = element
            .find("table > tbody > tr")
            .at(0)
            .find("td")
            .at(2)
            .text();
        const intField2 = element
            .find("table > tbody > tr")
            .at(1)
            .find("td")
            .at(2)
            .text();
        const intField3 = element
            .find("table > tbody > tr")
            .at(2)
            .find("td")
            .at(2)
            .text();

        expect(parseInt(intField, 10)).toBeLessThanOrEqual(parseInt(intField2, 10));
        expect(parseInt(intField2, 10)).toBeLessThanOrEqual(parseInt(intField3, 10));
    });

    test("Sorting (DESC)", () => {
        const element = mount(table);
        const tableElement = element.find("table > tbody > tr");
        let intField = element
            .find("table > tbody > tr")
            .at(0)
            .find("td")
            .at(2)
            .text();
        let sortIntFieldElement = element.find("table > thead > tr th").at(2);

        // not sorted
        expect(intField).toEqual("1");
        expect(sortIntFieldElement.text().trim()).toEqual("ID");

        // sort DESC
        sortIntFieldElement.simulate("click");
        sortIntFieldElement.simulate("click");
        intField = element
            .find("table > tbody > tr")
            .at(0)
            .find("td")
            .at(2)
            .text();
        sortIntFieldElement = element.find("table > thead > tr th").at(2);
        expect(intField).toEqual("1000");
        expect(sortIntFieldElement.text()).toEqual("IDarrow_downward");

        // next 3 pages should also be in downward trend
        const nextButton = element.find(".pagination > li").at(7);
        nextButton.simulate("click");
        nextButton.simulate("click");
        nextButton.simulate("click");
        intField = element
            .find("table > tbody > tr")
            .at(0)
            .find("td")
            .at(2)
            .text();
        const intField2 = element
            .find("table > tbody > tr")
            .at(1)
            .find("td")
            .at(2)
            .text();
        const intField3 = element
            .find("table > tbody > tr")
            .at(2)
            .find("td")
            .at(2)
            .text();

        expect(parseInt(intField, 10)).toBeGreaterThanOrEqual(parseInt(intField2, 10));
        expect(parseInt(intField2, 10)).toBeGreaterThanOrEqual(parseInt(intField3, 10));
    });

    test("Callbacks", () => {
        const cbSort = jest.fn();
        const cbPage = jest.fn();
        const cbRecord = jest.fn();

        const element = mount(tableWithCallbacks(cbSort, cbRecord, cbPage));
        let tableElement = element.find("table > tbody > tr");
        expect(tableElement.length).toEqual(10);

        const sortElement = element.find("table > thead > tr > th").at(0);
        const nextPageElement = element.find(".pagination > li").at(7);
        element.find(".table-footer .dropdown-field").simulate("click");

        const showRecordsElement = element.find(".dropdown-elements li").at(2);
        showRecordsElement.simulate("mousedown");
        tableElement = element.find("table > tbody > tr");

        expect(tableElement.length).toEqual(100);
        expect(cbRecord.mock.calls.length).toEqual(1);

        sortElement.simulate("click");
        sortElement.simulate("click");
        sortElement.simulate("click");
        expect(cbSort.mock.calls.length).toEqual(3);

        nextPageElement.simulate("click");
        nextPageElement.simulate("click");
        nextPageElement.simulate("click");
        expect(cbPage.mock.calls.length).toEqual(3);
    });
});
