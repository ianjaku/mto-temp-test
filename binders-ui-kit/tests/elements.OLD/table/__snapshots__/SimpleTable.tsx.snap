// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table Table (snapshot) 1`] = `
<div>
  <div
    className="table-wrapper"
  >
    <table
      className="table"
    >
      <thead>
        <tr>
          <th
            onClick={[Function]}
          >
            Name
          </th>
          <th
            onClick={[Function]}
          >
            E-mail
          </th>
          <th
            onClick={[Function]}
          >
            ID
          </th>
          <th
            onClick={[Function]}
          >
            actions
          </th>
        </tr>
      </thead>
      <tbody>
        <tr>
          <td>
            John Doe1
          </td>
          <td>
            johndoe1@manual.to
          </td>
          <td>
            1
          </td>
          <td>
            <button>
              Test
            </button>
          </td>
        </tr>
        <tr>
          <td>
            John Doe2
          </td>
          <td>
            johndoe2@manual.to
          </td>
          <td>
            2
          </td>
          <td>
            <button>
              Test
            </button>
          </td>
        </tr>
        <tr>
          <td>
            John Doe3
          </td>
          <td>
            johndoe3@manual.to
          </td>
          <td>
            3
          </td>
          <td>
            <button>
              Test
            </button>
          </td>
        </tr>
        <tr>
          <td>
            John Doe4
          </td>
          <td>
            johndoe4@manual.to
          </td>
          <td>
            4
          </td>
          <td>
            <button>
              Test
            </button>
          </td>
        </tr>
        <tr>
          <td>
            John Doe5
          </td>
          <td>
            johndoe5@manual.to
          </td>
          <td>
            5
          </td>
          <td>
            <button>
              Test
            </button>
          </td>
        </tr>
      </tbody>
    </table>
  </div>
  <div
    className="table-footer"
  >
    <ul
      className="pagination"
    >
      <span>
        Page
      </span>
      <li
        className="disabled"
        onClick={undefined}
      >
        <span
          className="material-icons pagination-icon"
          color=""
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          style={
            Object {
              "color": "rgba(0, 0, 0, 0.87)",
              "display": "inline-block",
              "fontSize": 24,
              "muiPrepared": true,
              "position": "relative",
              "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
              "userSelect": "none",
            }
          }
        >
          keyboard_arrow_left
        </span>
        <span
          className="material-icons pagination-icon"
          color=""
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          style={
            Object {
              "color": "rgba(0, 0, 0, 0.87)",
              "display": "inline-block",
              "fontSize": 24,
              "muiPrepared": true,
              "position": "relative",
              "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
              "userSelect": "none",
            }
          }
        >
          keyboard_arrow_left
        </span>
        First
      </li>
      <li
        className="disabled"
        onClick={undefined}
      >
        <span
          className="material-icons pagination-icon"
          color=""
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          style={
            Object {
              "color": "rgba(0, 0, 0, 0.87)",
              "display": "inline-block",
              "fontSize": 24,
              "muiPrepared": true,
              "position": "relative",
              "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
              "userSelect": "none",
            }
          }
        >
          keyboard_arrow_left
        </span>
        Previous
      </li>
      <li
        className="active"
        onClick={[Function]}
      >
        1
      </li>
      <li
        className=""
        onClick={[Function]}
      >
        2
      </li>
      <li
        className=""
        onClick={[Function]}
      >
        3
      </li>
      <li
        className=""
        onClick={[Function]}
      >
        4
      </li>
      <li
        className=""
        onClick={[Function]}
      >
        5
      </li>
      <li
        className=""
        onClick={[Function]}
      >
        Next 
        <span
          className="material-icons pagination-icon"
          color=""
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          style={
            Object {
              "color": "rgba(0, 0, 0, 0.87)",
              "display": "inline-block",
              "fontSize": 24,
              "muiPrepared": true,
              "position": "relative",
              "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
              "userSelect": "none",
            }
          }
        >
          keyboard_arrow_right
        </span>
      </li>
      <li
        className=""
        onClick={[Function]}
      >
        Last
        <span
          className="material-icons pagination-icon"
          color=""
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          style={
            Object {
              "color": "rgba(0, 0, 0, 0.87)",
              "display": "inline-block",
              "fontSize": 24,
              "muiPrepared": true,
              "position": "relative",
              "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
              "userSelect": "none",
            }
          }
        >
          keyboard_arrow_right
        </span>
        <span
          className="material-icons pagination-icon"
          color=""
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          style={
            Object {
              "color": "rgba(0, 0, 0, 0.87)",
              "display": "inline-block",
              "fontSize": 24,
              "muiPrepared": true,
              "position": "relative",
              "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
              "userSelect": "none",
            }
          }
        >
          keyboard_arrow_right
        </span>
      </li>
    </ul>
    <div
      className="dropdown borders"
      style={
        Object {
          "width": "100px",
        }
      }
    >
      <ul
        className="dropdown-field"
        onBlur={[Function]}
        onClick={[Function]}
        tabIndex={0}
      >
        <li
          className="dropdown-field-label"
          onMouseDown={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
        >
          <div
            className="dropdown-field-label-group"
          >
            <span
              className="dropdown-field-label-group-text"
            >
              10
            </span>
          </div>
          <span
            className="material-icons dropup-arrow"
            color=""
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            style={
              Object {
                "color": "rgba(0, 0, 0, 0.87)",
                "display": "inline-block",
                "fontSize": 24,
                "muiPrepared": true,
                "position": "relative",
                "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                "userSelect": "none",
              }
            }
          >
            arrow_drop_up
          </span>
        </li>
      </ul>
    </div>
  </div>
</div>
`;
